### PR TITLE
internalcontroller not needed since controller-runtime 0.6.2

### PR DESCRIPTION
**Pull request by joelddiaz**
Ever since the upgrade to controller-runtime 0.6.2 in
https://github.com/openshift/cloud-credential-operator/pull/232, we have
had the ability to slip in our own rateLimitter when building the
controller w/o needing to resort to our custom/copied internalcontroller.

Make use of this new ability of controller-runtime and retire our copy
of the internal controller implementation from controller-runtime.

This should make upgrading to future versions of controller-runtime a
bit easier.

xref: https://issues.redhat.com/browse/CCO-150

### CODE REVIEW

**Comment by 2uasimojo** [comment]
Oh, how lovely.

Just needs a `go mod tidy` I believe.

**Comment by joelddiaz** [comment]
/assign @lwan-wanglin @jeana-redhat @sferich888 
This should be a no-op in terms of features, functionality, UX as we are just starting to use some pre-existing features of our vendored version of controller-runtime so that we can drop some unnecessary copied code from older versions of controller-runtime.

File: pkg/operator/credentialsrequest/credentialsrequest_controller.go [hunk 1]
```diff
@@ -112,27 +112,18 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
...
-		Config:                  mgr.GetConfig(),
-		Scheme:                  mgr.GetScheme(),
-		Client:                  mgr.GetClient(),
-		Recorder:                mgr.GetEventRecorderFor(name),
-		Queue:                   workqueue.NewNamedRateLimitingQueue(rateLimiter, name),
-		MaxConcurrentReconciles: 1,
-		Name:                    name,
-	}
-
-	if err := mgr.Add(c); err != nil {
```
Comment (joelddiaz) [inline pkg/operator/credentialsrequest/credentialsrequest_controller.go:128]: Needed to remove this mgr.Add() as it is already called in the controller.New() codepath right above. Without this CCO errors early on startup with `time="2021-11-01T16:28:09Z" level=fatal msg="unable to run the manager" error="controller was started more than once. This is likely to be caused by being added to a manager multiple times"`

**Comment by 2uasimojo** [comment]
/retest

timeout

/lgtm

still

**Comment by lwan-wanglin** [comment]
do some tests, no regression. https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Runner/209292/console

/label qe-approved

**Comment by joelddiaz** [comment]
@2uasimojo PTAL, had to rebase due to other code changes. no functional changes from last time

**Comment by 2uasimojo** [comment]
Will restamp once CI is clear.

**Comment by joelddiaz** [comment]
Just internal bookkeeping changes with no change in product behaivor.
/label px-approved

**Pull request merged**